strats.md

```diff
@@ -454,3 +513,103 @@ A `comeInWithDoorStuckSetup` condition must match with a `leaveWithRunway` condi
+* _morphed_: A boolean indicating whether Samus must enter the room in a morphed state. This can be satisfied by obtaining or already being in an artificially morphed state, or by having collected the Morph item.
+* _mobility_: Takes one of three possible values, "mobile", "immobile", or "any", indicating whether or not Samus is
+required to be mobile (or immobile) after entering the room. The default value is "any". When entering with indirect G-mode, Samus is always mobile. With direct G-mode, Samus can be mobile if she takes knockback through the door transition and the reserve energy is low enough (<= 4 energy) so that knockback frames do not expire until after reserves finish filling.
+
+A `comeInWithGMode` entrance condition must match with either a `leaveWithGModeSetup` or `leaveWithGMode` entrance condition on the other side of the door:
+- If `mode` is "direct", then it can only match with a `leaveWithGModeSetup`. If `mode` is "indirect", then it can only match with a `leaveWithGMode`.
+
+When matching with a `leaveWithGModeSetup`, a `comeInWithGMode` has implicit requirements:
+- The tech requirement `canEnterGMode`.
+- The `XRayScope` item requirement.
+- A requirement to have at least 1 reserve energy.
+- A requirement to damage down to 0 energy, triggering reserves (causing the reserve energy to become zero and the regular energy to become what the reserve energy was).
+- The requirement `{"or": ["Morph", "canArtificialMorph"]}`, if the `morphed` property of `comeInWithGMode` is true.
+- Requirements for regaining mobility (there may be multiple options here, and they should be treated either as separate strats or the requirements should be joined as though with an `or`):
+  - G-mode mobile: Samus uses the knockback from the previous room to retain mobility. Here the `knockback` property of the `leaveWithGModeSetup` must be true, and reserve energy is assumed to have been drained to at most 4 before doing the strat.
+  - G-mode immobile: Samus uses knockback from a hit in the current room to regain mobility. The possibility of doing this is indicated by the presence of a strat with a `gModeRegainMobility` property with `from` and `to` node matching the `from` node (entrance door node) of the current strat. Any requirements of the `gModeRegainMobility` strat should be included as requirements to execute the current strat; it is important that these requirements be applied *after* the requirement to damage down to 0 energy and trigger reserves. If there are multiple such `gModeRegainMobility` strats, then these should be treated as multiple options.
```

Comment:
`from` and `to` matching the `from` sounds a bit weird since `from` will always be matching itself.

Reply:
This sounds odd and is very confusing to me. Maybe something along these lines would be more clear:
'''
... with a `from` and `to` node matching the entrance door node, as well as the `from` node of the current strat.
'''